### PR TITLE
Introduce VersionVector to detect the relationship between changes

### PR DESCRIPTION
**Pull request by hackerwins**
**What this PR does / why we need it**:

This PR introduces `VersionVector` for `Change.ID` to differentiate between Concurrent and Causal relationships among changes.

Existing Functionality of `Change.ID`:
- Allows for determining the Total Order of Changes.
- Generates `TimeTicket` using a provided `Delimiter`.
  - Used as an `ID` for Elements.
  - Utilized for Conflict Resolution of Operations.
  - Only establishes order in Total Order, unable to distinguish between Concurrent and Causal relationships.

New Features of `Change.ID`:
- Enables differentiation between Concurrent and Causal relationships among Changes.

We still use `LamportClock` in conjunction with VersionVector to reduce the size of CRDT metadata in `Document`s.

**Which issue(s) this PR fixes**:

Related to https://github.com/yorkie-team/yorkie/issues/723, https://github.com/yorkie-team/yorkie/pull/786

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

## Release Notes

- **New Features**
  - Introduced version vector handling for improved document synchronization and state management.
  - Added new fields for version vectors in various structures, enhancing the tracking of document changes.
  - Enhanced API responses to include version vectors, improving data integrity in collaborative environments.

- **Bug Fixes**
  - Improved error handling across multiple methods related to version vector operations.

- **Documentation**
  - Updated OpenAPI specifications to reflect new version vector schemas and properties.

- **Tests**
  - Expanded test coverage for version vector functionalities and garbage collection mechanisms.

### CODE REVIEW

**Comment by hackerwins** [comment]
@coderabbitai resume

**Comment by hackerwins** [comment]
@coderabbitai resume

File: pkg/document/change/id.go [hunk 1]
```diff
@@ -83,24 +97,44 @@ func (id ID) NewTimeTicket(delimiter uint32) *time.Ticket {
...
 }
 
-// SyncLamport syncs lamport timestamp with the given ID.
-// https://en.wikipedia.org/wiki/Lamport_timestamps#Algorithm
-func (id ID) SyncLamport(otherLamport int64) ID {
+// SyncClocks syncs logical clocks with the given ID.
+func (id ID) SyncClocks(other ID) ID {
+	lamport := id.lamport + 1
+	if id.lamport < other.lamport {
+		lamport = other.lamport
```
Comment (JOOHOJANG) [inline pkg/document/change/id.go:104]: I have a question about this function. Is the version vector being used here Lamport-synced?

If so, shouldn't the update logic be `other.lamport + 1` instead?

Comment (JOOHOJANG) [reply pkg/document/change/id.go:104]: Also, I think `new ID` should consider clients other than these two.
If I'm wrong, or if it seems like we're thinking of different ideas, please let me know.

**Pull request closed without merging**